lib/src/revset.pest

```diff
@@ -105,11 +105,10 @@ expression = {
   ~ (whitespace* ~ infix_op ~ whitespace* ~ (negate_op ~ whitespace*)* ~ range_expression)*
 }
 
-program = _{ SOI ~ whitespace* ~ expression ~ whitespace* ~ EOI }
-program_modifier = { identifier ~ pattern_kind_op ~ !":" }
-program_with_modifier = _{
+program = _{
   SOI ~ whitespace* ~ (program_modifier ~ whitespace*)? ~ expression ~ whitespace* ~ EOI
 }
+program_modifier = { identifier ~ pattern_kind_op ~ !":" }
```

Comment:
optional: we seem to mostly (only?) refer to rules defined earlier, so define this before `program`?

Reply:
Reordered. (maybe I messed up some of them before ;)